src/main/scala/rocket/DCache.scala

```diff
@@ -626,9 +626,10 @@ class DCacheModule(outer: DCache) extends HellaCacheModule(outer) {
   ccover(tl_out.d.valid && !tl_out.d.ready, "BLOCK_D", "D$ D-channel blocked")
 
   // Handle an incoming TileLink Probe message
-  val block_probe = releaseInFlight || grantInProgress || blockProbeAfterGrantCount > 0 || lrscValid
-  metaArb.io.in(6).valid := tl_out.b.valid && (!block_probe || lrscBackingOff)
-  tl_out.b.ready := metaArb.io.in(6).ready && !block_probe && !s1_valid && !s2_valid
+  val block_probe_for_core_progress = blockProbeAfterGrantCount > 0 || lrscValid
+  val block_probe_for_ordering = releaseInFlight || release_ack_wait || grantInProgress
+  metaArb.io.in(6).valid := tl_out.b.valid && (!block_probe_for_core_progress || lrscBackingOff)
+  tl_out.b.ready := metaArb.io.in(6).ready && !(block_probe_for_core_progress || block_probe_for_ordering || s1_valid || s2_valid)
```

Comment:
I don't understand why the conditions for metaArg.io.in(6).valid don't match thoes for tl_outb.ready.

Reply:
The arbiter valid signal is high more often than a probe can actually be accepted, as a means of preventing the core (which is the next-lowest priority port into the arbiter) from being able to initiate new accesses when a probe is pending.  This prevents a circular-wait condition, because probes can't be accepted while the cache pipeline is busy.

Reply:
Ahh. This makes sense. Thanks.